cli/internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,102 @@
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/rudderlabs/analytics-go/v4"
+	"github.com/rudderlabs/rudder-iac/cli/internal/config"
+)
+
+var (
+	once              sync.Once
+	v                 string
+	telemetryDisabled bool
+)
+
+func getEnvWithFallback(key, fallback string) string {
+	if value := os.Getenv(key); value != "" {
+		return value
+	}
+	return fallback
+}
+
+func Initialise(version string) {
+	once.Do(func() {
+		conf := config.GetConfig()
+
+		v = version
+		if conf.Telemetry.Disabled {
+			telemetryDisabled = true
+			return
+		}
+
+		disabled := getEnvWithFallback("RUDDER_CLI_TELEMETRY_DISABLED", "")
+		if disabled != "" && disabled == "1" {
+			telemetryDisabled = true
+			return
+		}
+
+		if conf.Telemetry.UserID == "" {
+			userID := uuid.New().String()
+			config.SetTelemetryUserID(userID)
+		}
+	})
+}
+
+func DisableTelemetry() {
+	telemetryDisabled := true
+	config.SetTelemetryDisabled(telemetryDisabled)
+}
+
+func EnableTelemetry() {
+	telemetryDisabled := false
+	config.SetTelemetryDisabled(telemetryDisabled)
+}
+
+func track(event string, properties analytics.Properties) error {
+	if telemetryDisabled {
```

Comment:
Why not just `if config.Telemetry.Disabled` ?

Reply:
Telemetry can also be disabled through an env flag with gets read on the Initialise command in L36. So config is not the only way to disable the telemetry and this might be used in the github action where we don't wanna run the disable command everytime on the cli and can control it through the ENV.

Let me know if my approach makes sense to you ?

Reply:
The idea that config is used as the source of truth of configuration for the components, and it's using viper to merge local file configurations with configurations coming from envs. This is how the access token is handled as well. It can be set either way, but [read directly from config ](https://github.com/rudderlabs/rudder-iac/blob/18fda4c48382b4a0da49639780f95499bc270e98/cli/internal/app/dependencies.go#L52).

The only problem with this is if we initialise the telemetry component before we initialize config, but in this case I understand we initialize it afterwards.

Reply:
Make sense, I'll refactor this then.